### PR TITLE
react: Added InvalidEvent, onInvalid & onInvalidCapture

### PR DESCRIPTION
**Pull request by shangxiao**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://developer.mozilla.org/en-US/docs/Web/Events/invalid
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

A few more things:
 - tsc is failing on lines that I have not modified
 - invalid is an "Uncategorized" event [according to MDN](https://developer.mozilla.org/en-US/docs/Web/Events#Uncategorized_events), I've bundled it with the other form events, there doesn't seem to be a strict categorisation here

### CODE REVIEW

**Comment by vsiao** [comment]
Based on https://github.com/facebook/react/blob/b1b4a2fb252f26fe10d29ba60d85ff89a85ff3ec/src/renderers/dom/shared/eventPlugins/SimpleEventPlugin.js#L167 you should use a simple `ReactEventHandler<T>` to type `invalid` events.

**Comment by rapilabs** [comment]
@vsiao just looking at that react code… it looks like the other form events are categorised similarly.  Note: I'm not familiar with react internals and I'm only quite new to ts myself so I guess I'll just take your lead on this ;)

**Comment by vsiao** [comment]
That's true. The event types that we specify in the typings generally mirror the ones listed in the docs at https://facebook.github.io/react/docs/events.html#form-events. I noticed that onInvalid isn't listed on that page either, so you might want to send a PR over there as well :)

File: types/react/index.d.ts [hunk 1]
```diff
@@ -330,6 +330,10 @@ declare namespace React {
     interface FormEvent<T> extends SyntheticEvent<T> {
     }
 
+    interface InvalidEvent<T> extends SyntheticEvent<T> {
+        target: EventTarget & T;
+    }
+
```
Comment (vsiao) [inline types/react/index.d.ts]: did you mean to leave this here?

Comment (rapilabs) [inline types/react/index.d.ts]: @vsiao oh uh I wasn't expecting this to be merged until your suggestion was applied :)  but yep this was intentional to gain the target attribute.

**Comment by vsiao** [comment]
@minestarks could you wait for explicit approval before merging PRs in the future? thanks.

**Pull request merged**